### PR TITLE
Compression

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

File: draft-rescorla-tls-ctls.md [hunk 1]
```diff
@@ -31,14 +37,14 @@ informative:
...
-be possible for the server to distinguish TLS 1.3 and CTLS handshakes.
+eventually be possible for the server to distinguish TLS 1.3 and CTLS handshakes.
 
 
 --- middle
 
 # Introduction
 
-DDISCLAIMER: This is a work-in-progress draft of MLS and has not yet
+DDISCLAIMER: This is a work-in-progress draft of cTLS and has not yet
```
Comment (bifurcation) [inline draft-rescorla-tls-ctls.md]: Nit: DDISCLAIMER

File: draft-rescorla-tls-ctls.md [hunk 2]
```diff
@@ -55,11 +61,10 @@ For the common (EC)DHE handshake with (EC)DHE and pre-established
...
 
-Although isomorphic, CTLS implementations cannot interoperate with TLS 1.3
-implementations because the packet formats are non-interoperable. It is
-probably possible to make a TLS 1.3 server switch-hit between CTLS and TLS 1.3
-but this specification does not define how.
-
+Because cTLS is semantically equivalent to TLS, it can be viewed either
+as a related protocol or as a compression mechanism. Specifically, it
+can be implemented by a layer between the TLS handshake state
+machine and the record layer. In order to enable this, the cTLS trans
```
Comment (bifurcation) [inline draft-rescorla-tls-ctls.md]: This ends prematurely.

File: draft-rescorla-tls-ctls.md [hunk 3]
```diff
@@ -506,6 +511,46 @@ Total: 113 + X bytes
...
+With very aggressive parameters, this technique can bring the
+handshake down to about XXX [[RLB]] bytes.
+
+The major potential disadvantage is that it makes cTLS and TLS
+handshakes confusable. For instance, an attacker who obtained the
+handshake keys might be able to undetectably transform a cTLS <-> TLS
+connection into a TLS <-> TLS connection. This is easily dealt with
+by modifying the transcript, e.g., by injecting a cTLS extension in
+the transcript (though not into cTLS wire format).
+
```
Comment (bifurcation) [inline draft-rescorla-tls-ctls.md]: Nit: Extra blank line

File: draft-rescorla-tls-ctls.md [hunk 4]
```diff
@@ -506,6 +511,46 @@ Total: 113 + X bytes
...
+security proofs even with very aggressive compression (with suitable
+proofs about the bijectiveness of the compression). [Thanks to Karthik
+Bhargavan for this point.] This probably also makes it easier to
+implement more aggressive compression. For instance, the above text
+shrinks the handshake headers but does not elide them entirely.
+If the handshake shape (i.e., which messages are sent) is known in
+advance, then these headers can be removed, thus trimming about 20 bytes
+from the handshake. This is easier to reason about as a form of compression.
+With very aggressive parameters, this technique can bring the
+handshake down to about XXX [[RLB]] bytes.
```
Comment (bifurcation) [inline draft-rescorla-tls-ctls.md]: XXX = 48 / 164 / 116 for M1 / M2 / M3, with:
* Mutual auth
* 16 byte random
* Finished included

File: draft-rescorla-tls-ctls.md (hunk 3 above)
Comment (ekr) [reply draft-rescorla-tls-ctls.md]: Appreciate the value you add here.

**Pull request merged**